### ISSUE TITLE
Best navigation manager

### ISSUE THREAD

**Issue by carmas123** [lulupointu/vrouter#186]
Hi all, I try several navigation manager from the 2.0 navigation system of Flutter and I think that this is the best library over all.
I disappoint the Flutter comunity to choice GoRouter as internal library for flutter some days ago.
In GoRouter missing many features of VRouter and I'm very hangry for that!!

Now the problem is that many lib developed by a single developer is hard to maintain for a long time and sooner or later also this lib will also end.

Please ask all of you for the best developer support and help to keep this wonderful library up and running.

Very good job @lulupointu

**Comment by lulupointu** [comment]
Thanks for the kind message, it always helps staying motivated.

I was wondering, could you tell me the issues that you encountered in `go_router` that you found solutions in `vrouter`?

**Comment by carmas123** [comment]
Primarily is the handling of guards: in VRourer it is done as it should be done.
This is crucial for me.
In VRouter we can handle the routing with async methods while in go_router no.

However I think that when you make such a decision you shouldn't do it just because one lib has more likes than another.

The potential of one and the other should be analyzed as a whole.

Now my question is: what does go_router have that VRouter doesn't have?

💪

**Comment by TatsuUkraine** [comment]
@carmas123 ability to push one single route in the stack without the need to rebuild entire history))

**Comment by alphamikle** [comment]
I totally agree with opinion, that `vrouter` better then `go_router`. Just now I creating a new application and had experiments with both of them and approach of `vrouter` seems like more user-friendly and more feature-riched. Only one thing, which must need to be implemented in `vrouter` - is ability to push routes in imperative style. That might be helpful for some kind of dynamic routing when you can open something from any screen without additional nesting or stacking.
